lab6/react-spa/src/components/main/main.jsx

```diff
@@ -0,0 +1,385 @@
+import React, {useEffect, useMemo, useState} from 'react';
+import {useAuth} from "../all/api/useAuth";
+import Load from "../loading/loading";
+import {get_current_weather} from "../all/api/openweatherapi";
+import {change_wallPaper, phenomenons} from "./Paper_changer";
+
+const Main = ()=> {
+    const { isSignedIn} = useAuth();
+    const var1 = useMemo(()=>"some",[]);
+    const [date, setDate] = useState("");
+    const [month, setMonth] = useState("");
+    const [day, setDay] = useState("");
+    const [per, setPer] = useState("0%");
+    const [pres, setPres] = useState("23");
+    const [wind_str, setWstr] = useState("0MPS");
+    const [state, setState] = useState('None');
+    const [temp, setTemp] = useState('0' +  '&deg;');
+    let months = ["January", "February", "March", "April", "May", "June",
+        "July", "Augest", "September", "October", "November", "December"];
+//###########################################################
+    function update_weather(current)
+    {
+        update_widget_temp(current.temperature,current.state)
+        update_widget_humidity(current.humidity)
+        update_pressure_widget(current.pressure)
+        update_wind_widget(current.wind_str,current.wind_dir)
+        change_wallPaper(current.state)
+    }
+
```

Comment:
all ur widgets better be a components...

Reply:
created component for each widget